### ISSUE TITLE
Ask for Autofill permission with Xiaomi

### ISSUE THREAD

**Issue by J-Jamet** [Kunzisoft/KeePassDX#629]
Request real-time permissions for MIUI phones. Unfortunately I can not test because I do not have these devices so if charitable souls can give me feedback once the code is created, it would be very appreciated.

**Comment by J-Jamet** [comment]
https://stackoverflow.com/questions/36123349/how-to-open-draw-overlay-permission-popup-in-miui

**Comment by glemco** [comment]
I just got a xiaomi phone and was surprised the autofill wasn't working, before checking here.
Just tell me what and how should I check!

**Comment by J-Jamet** [comment]
At the moment, It is possible to run autofill for Android devices but you have to activate permissions manually.
https://github.com/Kunzisoft/KeePassDX/issues/428
Since I don't have a Xiaomi device yet, I cannot code a feature to request permission to be enabled. (I just ordered a used device online)
It is also possible that it depends on the version of IMUI, so I will develop the dialog but it will be necessary to test later.

**Comment by glemco** [comment]
Thanks, I figured out I had to enable that settings, anyway if you need the feature to be checked under miui 12 just make a whistle!

**Comment by J-Jamet** [comment]
I just bought a used Xiaomi device and it's a pain to start developing with this OS. I have to create a Mi account and connect to the internet to activate the developer options and start working locally from a USB cable ...

**Comment by J-Jamet** [comment]
I have tried a lot of things and the big problem is that we can't call permissions natively because it's in the MIUI hardcode. 
One of the solutions is to do code introspection but I don't want to use this method because we need to create an introspection method for each version of MIUI. Which is horrible to maintain and debug.
It's just very poorly handled by Xiaomi which blocks too many things. I have found that disabling MIUI optimization fixes all the issues so I will add this section to the wiki.
